### ISSUE TITLE
Removal of deep copy functions breaks casskop as dependency in other projects

### ISSUE THREAD

**Issue by jsanda** [Orange-OpenSource/casskop#181]
I am import casskop as a dependency in some projects. In one project I am currently using v0.3.2 and decided to upgrade to v0.5.0-release. I updated my `go.mod` and then the build failed with:

```
$ make build-image
operator-sdk generate k8s
INFO[0000] Running deepcopy code-generation for Custom Resource group versions: [thelastpickle:[v1alpha1], ]
INFO[0006] Code-generation complete.
operator-sdk generate openapi
INFO[0000] Running OpenAPI code-generation for Custom Resource group versions: [thelastpickle:[v1alpha1], ]
INFO[0015] Created deploy/crds/thelastpickle_v1alpha1_tlpstress_crd.yaml
INFO[0023] Created deploy/crds/thelastpickle_v1alpha1_tlpstresscontext_crd.yaml
INFO[0023] Code-generation complete.
# github.com/Orange-OpenSource/cassandra-k8s-operator/pkg/apis/db/v1alpha1
../../../../pkg/mod/github.com/!orange-!open!source/cassandra-k8s-operator@v0.5.0-release/pkg/apis/db/v1alpha1/cassandracluster_types.go:162:8: lastcc.Status undefined (type *"k8s.io/apimachinery/pkg/apis/meta/v1".ObjectMeta has no field or method Status)
../../../../pkg/mod/github.com/!orange-!open!source/cassandra-k8s-operator@v0.5.0-release/pkg/apis/db/v1alpha1/cassandracluster_types.go:400:7: newcc.InitCassandraRackList undefined (type *"k8s.io/apimachinery/pkg/apis/meta/v1".ObjectMeta has no field or method InitCassandraRackList)
../../../../pkg/mod/github.com/!orange-!open!source/cassandra-k8s-operator@v0.5.0-release/pkg/apis/db/v1alpha1/cassandracluster_types.go:404:20: newcc.Status undefined (type *"k8s.io/apimachinery/pkg/apis/meta/v1".ObjectMeta has no field or method Status)
../../../../pkg/mod/github.com/!orange-!open!source/cassandra-k8s-operator@v0.5.0-release/pkg/apis/db/v1alpha1/cassandracluster_types.go:874:25: cannot use &CassandraCluster literal (type *CassandraCluster) as type "k8s.io/apimachinery/pkg/runtime".Object in argument to SchemeBuilder.Register:
	*CassandraCluster does not implement "k8s.io/apimachinery/pkg/runtime".Object (missing DeepCopyObject method)
../../../../pkg/mod/github.com/!orange-!open!source/cassandra-k8s-operator@v0.5.0-release/pkg/apis/db/v1alpha1/cassandracluster_types.go:874:46: cannot use &CassandraClusterList literal (type *CassandraClusterList) as type "k8s.io/apimachinery/pkg/runtime".Object in argument to SchemeBuilder.Register:
	*CassandraClusterList does not implement "k8s.io/apimachinery/pkg/runtime".Object (missing DeepCopyObject method)
Error: failed to build operator binary: (failed to exec []string{"go", "build", "-o", "/Users/jsanda/Development/gocode/src/github.com/jsanda/tlp-stress-operator/build/_output/bin/tlp-stress-operator", "-gcflags", "all=-trimpath=/Users/jsanda/Development/gocode/src/github.com/jsanda", "-asmflags", "all=-trimpath=/Users/jsanda/Development/gocode/src/github.com/jsanda", "github.com/jsanda/tlp-stress-operator/cmd/manager"}: exit status 2)
Usage:
  operator-sdk build <image> [flags]

Flags:
      --go-build-args string      Extra Go build arguments as one string such as "-ldflags -X=main.xyz=abc"
  -h, --help                      help for build
      --image-build-args string   Extra image build arguments as one string such as "--build-arg https_proxy=$https_proxy"
      --image-builder string      Tool to build OCI images. One of: [docker, podman, buildah] (default "docker")

Global Flags:
      --verbose   Enable verbose logging

make: *** [build-image] Error 1
```
My project is using a newer version of operator-sdk. I thought that might be a problem; so, I reverted to an earlier version of my project which was using operator-sdk v0.9.0 and still got the error.

After some investigation it looked like the issue was due to commit 2c40d70658b9bf6f which deleted the generated deep copy functions. 

In my fork of casskop I created a branch off of master and committed the deep copy functions. I then updated my project to point to my fork of casskop and it built without error.

**Comment by allamand** [comment]
Hi John, okay this makes sense to keep deep copy functions for dependencies

**Comment by cscetbon** [comment]
Fixed by #195
